File: leetcode523/solveLeetcode523.cpp
// class Solution 
// {
// public:
//     bool checkSubarraySum(vector<int>& nums, int k) 
//     {
//         const int n=nums.size();
//         if(n<2)return false;

//         unordered_map<int,int>hash;
//         hash[0]=-1;


//         int i,reminder=0,preIndex;
//         for(i=0;i<n;++i)
//         {
//             reminder=(reminder+nums[i])%k;
//             if(hash.count(reminder))
//             {
//                 preIndex=hash[reminder];
//                 if(i-preIndex>=2)return true;
//             }
//             else hash[reminder]=i;
            
//         }

//         return false;
//     }
// };

File: DataStructure/prefix/leetcode523/solveLeetcode523.go
// func checkSubarraySum(nums []int, k int) bool {
//     n:=len(nums)
//     if n<2{
//         return false
//     }

//     f:=map[int]int{0:-1}

//     //var preIndex int
//     remainder:=0

//     for i:=0;i<n;i++{
//         remainder=(remainder+nums[i])%k

//         if preIndex,has:=f[remainder];has{
//             preIndex=f[remainder]
//             if i-preIndex>=2{
//                 return true
//             }
//         }else{
//             f[remainder]=i
//         }
//     }

//     return false
// }